src/java/org/pantsbuild/tools/junit/impl/ScalaTestUtil.java

```diff
@@ -30,11 +44,6 @@ public static Runner getJUnitRunner(Class<?> clazz) {
    * @return true if the test class is a scalatest test, false if not.
```

Comment:
You should probably also update the javadoc because we're no longer using the test classes class loader.

I'm not sure if it is worth it, but it might make sense to assert that the clazz classloader is the same as the suiteClass's or is in the same loader hierarchy. I think Stu's right that we just use one, but this feels like a place that could be forgotten if that gets changed in the future. A test case that would fail if they were different would also work for me.

Reply:
Ok I'll update the doc and look into adding a unit test.

Reply:
In regards to 

> assert that the clazz classloader is the same as the suiteClass's

I was able to do this as a sanity check locally, but in the long run I think if something broke then `suiteClass` wouldn't be loaded and would be null, making the assert tricky.

I think there is enough coverage around this code in `UtilTest` and `SpecParserTest` that a test would fail if something were broken, but I don't know how to test the class loading directly.

Any suggestions on where/how I could test this?

Reply:
I think the way I'd approach it is to create a one-off classloader that has ScalaSuite and a subclass of ScalaSuite loaded into it, then assert that passing the subclass into the util method fails in a understandable way. Constructing that test might be a bit tricky if you're not familiar with classloaders.

My guess for the failure mode is that the separate classloader owned ScalaSuite would fail assignFrom and wouldn't be considered a scalatest. If that's the case the user facing behavior would be that the test would either not be run or would fail to run. But, I'm not sure.